### PR TITLE
Improve List IDP endpoint with addtional query param called ''requiredAttributes" 

### PR DESCRIPTION
**Pull request by sarubi**
## Purpose
Fix this wso2/product-is#7823 issue in IDP REST API
Resolves wso2/product-is#7864

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/resources/idp.yaml [hunk 1]
```diff
@@ -1533,6 +1532,28 @@ components:
         self:
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/resources/idp.yaml]: let's add this attribute as the last in the object definition

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java [hunk 2]
```diff
@@ -110,22 +110,26 @@
...
-    public IdentityProviderListResponse getIDPs(String attributes, Integer limit, Integer offset, String filter, String
-            sortBy, String sortOrder) {
+    public IdentityProviderListResponse getIDPs(String requiredAttributes, Integer limit, Integer offset, String filter,
+                                                String sortBy, String sortOrder) {
 
-        handleNotImplementedCapabilities(attributes);
         try {
+            List<String> requestedAttributeList = null;
+            if (StringUtils.isNotBlank(requiredAttributes)) {
+                requestedAttributeList = new ArrayList<>(Arrays.asList(requiredAttributes.split(",")));
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java]: we can use StringUtils.split and void if (StringUtils.isNotBlank(requiredAttributes))  check

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java [hunk 3]
```diff
@@ -1193,6 +1190,57 @@ private IdentityProviderListResponse createIDPListResponse(IdpSearchResult idpSe
...
+                        break;
+                    case Constants.ROLES:
+                        identityProviderListItem.setRoles(createRoleResponse(idp));
+                        break;
+                    case Constants.FEDERATED_AUTHENTICATORS:
+                        identityProviderListItem.setFederatedAuthenticators(createFederatedAuthenticatorResponse(idp));
+                        break;
+                    case Constants.PROVISIONING:
+                        identityProviderListItem.setProvisioning(createProvisioningResponse(idp));
+                        break;
```
Comment (sarubi) [inline components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java]: default case and throw exception

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java (hunk 2 above)
Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java]: StringUtils.split(requiredAttributes, ",") will return null if requiredAttributes is null. 
Arrays.asList(StringUtils.split(requiredAttributes, ",")) will thow NPE if StringUtils.split(requiredAttributes, ",") return null, so we need to add null check.

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/resources/idp.yaml (hunk 1 above)
Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/resources/idp.yaml]: fixed

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApiService.java [hunk 4]
```diff
@@ -20,6 +20,7 @@
 import org.wso2.carbon.identity.api.server.idp.v1.model.*;
 import org.apache.cxf.jaxrs.ext.multipart.Attachment;
 import org.apache.cxf.jaxrs.ext.multipart.Multipart;
+import java.io.InputStream;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApiService.java]: do we need this import?

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApi.java [hunk 5]
```diff
@@ -17,6 +17,10 @@
 package org.wso2.carbon.identity.api.server.idp.v1;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApi.java]: do we need these imports?

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApiService.java (hunk 4 above)
Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApiService.java]: this is generated code, do we need to alter it?

File: components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApi.java (hunk 5 above)
Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/gen/java/org/wso2/carbon/identity/api/server/idp/v1/IdentityProvidersApi.java]: this also under generated code..

**Pull request merged**